tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Test_KerrSchild.cpp

```diff
@@ -0,0 +1,187 @@
+  const auto& dt_lapse = get<EinsteinSolutions::KerrSchild::dt_lapse>(vars);
+  const auto& d_lapse = get<EinsteinSolutions::KerrSchild::deriv_lapse>(vars);
+  const auto& shift = get<EinsteinSolutions::KerrSchild::shift>(vars);
+  const auto& d_shift = get<EinsteinSolutions::KerrSchild::deriv_shift>(vars);
+  const auto& dt_shift = get<EinsteinSolutions::KerrSchild::dt_shift>(vars);
+  const auto& g = get<EinsteinSolutions::KerrSchild::spatial_metric>(vars);
+  const auto& dt_g =
+      get<EinsteinSolutions::KerrSchild::dt_spatial_metric>(vars);
+  const auto& d_g =
+      get<EinsteinSolutions::KerrSchild::deriv_spatial_metric>(vars);
+
+  // Check those quantities that should be zero.
+  const auto zero = make_with_value<DataVector>(x, 0.);
+  CHECK(dt_lapse.get() == zero);
+  for (size_t i = 0; i < 3; ++i) {
+    CHECK(dt_shift.get(i) == zero);
+    for (size_t j = 0; j < 3; ++j) {
+      CHECK(dt_g.get(i, j) == zero);
+    }
+  }
+
+  const DataVector r = magnitude(x);
+  auto lapse_test = make_with_value<Scalar<DataVector>>(x, 0.0);
+  get<>(lapse_test) = 1.0 / sqrt(1.0 + 2.0 * mass / r);
+  CHECK_ITERABLE_APPROX(lapse, lapse_test);
+
+  auto shift_test =
+      make_with_value<tnsr::I<DataVector, 3, Frame::Inertial>>(x, 0.0);
+  for (size_t i = 0; i < 3; ++i) {
+    shift_test.get(i) =
+        2.0 * mass * x.get(i) / square(r) * square(get<>(lapse));
+  }
+  CHECK_ITERABLE_APPROX(shift, shift_test);
+
+  auto d_lapse_test =
+      make_with_value<tnsr::i<DataVector, 3, Frame::Inertial>>(x, 0.0);
+  for (size_t i = 0; i < 3; ++i) {
+    d_lapse_test.get(i) = mass * x.get(i) / cube(r) * cube(get<>(lapse));
+  }
+  CHECK_ITERABLE_APPROX(d_lapse, d_lapse_test);
+
+  auto g_test =
+      make_with_value<tnsr::ii<DataVector, 3, Frame::Inertial>>(x, 0.0);
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = i; j < 3; ++j) {
+      g_test.get(i, j) = 2.0 * mass * x.get(i) * x.get(j) / cube(r);
+    }
+    g_test.get(i, i) += 1.0;
+  }
+  CHECK_ITERABLE_APPROX(g, g_test);
+
+  auto dg_test =
+      make_with_value<tnsr::ijj<DataVector, 3, Frame::Inertial>>(x, 0.0);
+  for (size_t k = 0; k < 3; ++k) {
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t j = i; j < 3; ++j) {  // Symmetry
+        dg_test.get(k, i, j) =
+            -6.0 * mass * x.get(i) * x.get(j) * x.get(k) / cube(r) / square(r);
+        if (k == j) {
+          dg_test.get(k, i, j) += 2.0 * mass * x.get(i) / cube(r);
+        }
+        if (k == i) {
+          dg_test.get(k, i, j) += 2.0 * mass * x.get(j) / cube(r);
+        }
+      }
+    }
+  }
+  for (size_t k = 0; k < 3; ++k) {
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t j = i; j < 3; ++j) {  // Symmetry
+        CHECK_ITERABLE_APPROX(d_g.get(k, i, j), dg_test.get(k, i, j));
+      }
+    }
+  }
+}
+
+void test_einstein_solution() {
+  // Parameters
+  //   ...for KerrSchild solution
+  const double mass = 1.01;
+  const std::array<double, 3> spin = {{0.1, 0.2, 0.3}};
```

Comment:
General question: is it any better to do `spin{{0.1, 0.2, 0.3}}` instead?

Reply:
The code has the same meaning, but I prefer `spin{{...}}}` because that makes it clear it is a constructor call, not an assignment operator call. This seems to be (understandably) confusing to people newer to C++. That is, they think because of the `=` it must use `operator=`, when really the compiler just calls the constructor.